packages/jest/src/automatic.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Options for Automatic checks to be passed to {@link registerSa11yAutomaticChecks}
+ */
+export type AutoCheckOpts = {
+    runAfterEach?: boolean;
+    cleanupAfterEach?: boolean;
+    // TODO (feat): add support for result consolidation across all tests for a test run
+    //  and optional exclusion of selected tests
+    // consolidateResults?: boolean;
+    // excludeTests?: string[];
+};
+
+/**
+ * Default options when {@link registerSa11yAutomaticChecks} is invoked
+ */
+const defaultAutoCheckOpts: AutoCheckOpts = {
+    runAfterEach: true,
+    cleanupAfterEach: true,
+};
+
+/**
+ * Run accessibility check on each element node in the body using {@link toBeAccessible}
+ * @param cleanup - boolean indicating if the DOM should be cleaned up after
+ */
+export async function automaticCheck(cleanup = false): Promise<void> {
+    // Create a DOM walker filtering only elements (skipping text, comment nodes etc)
+    const walker = document.createTreeWalker(document.body, NodeFilter.SHOW_ELEMENT);
+
+    try {
+        while (walker.nextNode()) await expect(walker.currentNode).toBeAccessible();
```

Comment:
Good call using a TreeWalker here. Using `nextNode()` will essentially do a depth first traversal of every element in the DOM. The original code and comment made it sound like you wanted to just iterate over the immediate children of the body. 

Iterating through every single element seems unnecessarily slow, no?

Reply:
Oops, thanks for catching that!
Yes I wanted to iterate over the immediate children. Fixed now.
Weird that there is no easier way to iterate through all children with tree walker.